### ISSUE TITLE
AttributeError: 'Blueprint' object has no attribute 'before_app_first_request'. Did you mean: 'before_app_request'?

### ISSUE THREAD

**Issue by EletrixtimeYT** [flask-dashboard/Flask-MonitoringDashboard#433]
LocalCache\local-packages\Python311\site-packages\flask_monitoringdashboard\__init__.py", line 67, in bind
    blueprint.before_app_first_request(init_measurement)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Blueprint' object has no attribute 'before_app_first_request'. Did you mean: 'before_app_request'?

**Comment by django-datama** [comment]
I think what should have been done here is to do something like this :


```diff
    from flask_monitoringdashboard.core import custom_graph

-    blueprint.before_app_first_request(init_measurement)
+    init_measurement()
-    blueprint.before_app_first_request(init_cache)
+    init_cache()

    if schedule:

```

**Comment by django-datama** [comment]
Do you want help on this @mircealungu or @FlorianRhiem (blame 😆)

I can gladely open a pull request solving this if you'd like.

**Comment by pascal-mueller** [comment]
This error occurs because in flask 2.3.0 they removed the two functions. See https://flask.palletsprojects.com/en/2.3.x/changes/#version-2-3-0

> The app.before_first_request and bp.before_app_first_request decorators are removed.

See also https://github.com/pallets/flask/issues/4605

**Comment by mircealungu** [comment]
Thanks @pascal-mueller ! Do you have an idea how to fix this? I do not have much capacity for the project at the moment :(

**Comment by mircealungu** [comment]
the issue should be fixed in FMD version [3.1.2](https://pypi.org/project/Flask-MonitoringDashboard/).  that was just released

**Comment by EvgenyTishkin** [comment]
this error stil exists! fix it idiots
